zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java

```diff
@@ -473,8 +474,23 @@ public Response runParagraph(@PathParam("notebookId") String notebookId,
       }
     }
 
-    note.run(paragraph.getId());
-    return new JsonResponse<>(Status.OK).build();
+    if (paragraph.getNoteReplLoader() == null) {
+      paragraph.setNoteReplLoader(note.getNoteReplLoader());
+    }
+
+    if (paragraph.getListener() == null) {
+      paragraph.setListener(note.getJobListenerFactory().getParagraphJobListener(note));
+    }
+
+    paragraph.run();
+
+    final InterpreterResult result = paragraph.getResult();
+    if (result.code() == InterpreterResult.Code.SUCCESS) {
+      return new JsonResponse<>(Status.OK, result.message()).build();
```

Comment:
How about return `result` instead of just `result.message`?
So user can access all the information of code, type, message in the result.